2주차 - 프로젝트 기획 및 DB 설계

tbnsok40 opened the pull request:
> #### 📕 이번 주가 지나면 사용자는 무엇을 할 수 있나요?
- Page API 를 모두 사용할 수 있습니다.

> #### 📕 이번 주에 구현할 기능을 우선순위대로 작성해 주세요.
- **Page 관련 주된 코드 작성**
1. PageController 완성
2. PageService 완성
3. Page DTO 완성
4. Page 테스트코드 작성

tbnsok40 committed:
PageService, PageRepository 파일 생성
CrudRepository 를 상속받은 PageRepository 생성

hannut91 commented:
Page API를 사용할 수 있다고 하셨는데, 사용자는 어떤 가치를 얻을 수 있나요? 어떤 혜택일 위해서 해당 API를 사용하나요?

hannut91 commented:
테스트를 먼저 작성하여야합니다. 코드를 작성하고 나서 테스트를 작성하는 것은 기존 코드가 동작하는 것을 자동으로 확인해줄 코드를 작성하는 것으로 의미가 있습니다. 하지만 더 나아가서 테스트를 먼저 작성하면 테스트를 설계의 도구 중 하나로 이용합니다. 내가 코드를 작성하기 전에 요구사항을 먼저 정의하고, 어떻게 코드를 사용할지 정의하기 때문입니다. 

따라서 테스트를 먼저 작성해보세요

tbnsok40 committed:
PageControllerTest 작성중
getOnePage 테스트 중 실패

tbnsok40 committed:
getOnePage 테스트 에러 해결
컨트롤러 메서드의 인자에 @PathVariable 어노테이션 추가

tbnsok40 committed:
PageController create 메소드 추가

tbnsok40 committed:
오타 수정

tbnsok40 committed:
getNotExistedPage 테스트 에러
NotFoundException 예외 발생

tbnsok40 committed:
PageController create 테스트 작성

tbnsok40 committed:
PageController update 메소드 및 테스트 추가, UserData 클래스 삭제
서버 실행 시, UserData 클래스에서 Jpa관련 에러가 발생하여 우선 삭제.
PageController 완성 후, User 관련 기능 재작성할 것

tbnsok40 commented:
> Page API를 사용할 수 있다고 하셨는데, 사용자는 어떤 가치를 얻을 수 있나요? 어떤 혜택일 위해서 해당 API를 사용하나요?

Page 는 사용자의 개인 페이지로 일종의 포트폴리올 개념이라고 할 수 있습니다. 이 점이 불명확한 것 같아, page 를 모두 portfolio로 수정을 고려하고 있습니다.

Page API 를 사용하면, 사용자는 개인 포트폴리오 페이지를 가질 수 있게 됩니다!

tbnsok40 commented:
안녕하세요 윤석님, 몇가지 질문이 있습니다!
1. 다음 주부터 프론트와 백엔드를 병행하며 개발할 계획인데, 혹시 이 때 어떤 점을 잘 고려해야 할지 궁금합니다!
2. Page 엔티티와 User 엔티티(새로 생성할 계획)를 1대1 매핑할 계획입니다. 각 User 마다 하나의 Page를 갖기 때문입니다. 이때 1대1 매핑에서 양방향 관계와 단반향 관계라는 것을 알게 됐는데, 이 둘의 명확한 차이를 이해하지 못하였습니다. 혹시 어떤 관계를 선택해야 하는지 궁금합니다!

tbnsok40 committed:
Page -> Portfolio 로 naming 수정

tbnsok40 committed:
postgresql 설치 전 no 에러 상태

tbnsok40 commented:
postgresql 을 연동하는 과정에서 아래와 같은 에러가 발생하였습니다.

` Error creating bean with name 'portfolioRepository' defined in project.spring.tbnsok40.domain.PortfolioRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Cannot resolve reference to bean 'jpaMappingContext' while setting bean property 'mappingContext';`

이해한 바로는, PortfolioRepository bean 을 생성하지 못해 발생한 에러입니다.
 PortfolioRepository 를 사용하는 PortfolioService 에서  PortfolioRepository의 선언부에 @Autowired  어노테이션을 붙여봤지만 동일한 에러메시지가 발생하여, 에러의 다른 원인이 무엇인지 질문드리고 싶습니다..!

johngrib commented:
> postgresql 을 연동하는 과정에서 아래와 같은 에러가 발생하였습니다.
> 
> ` Error creating bean with name 'portfolioRepository' defined in project.spring.tbnsok40.domain.PortfolioRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Cannot resolve reference to bean 'jpaMappingContext' while setting bean property 'mappingContext';`
> 
> 이해한 바로는, PortfolioRepository bean 을 생성하지 못해 발생한 에러입니다.
> PortfolioRepository 를 사용하는 PortfolioService 에서 PortfolioRepository의 선언부에 @Autowired 어노테이션을 붙여봤지만 동일한 에러메시지가 발생하여, 에러의 다른 원인이 무엇인지 질문드리고 싶습니다..!

이럴 땐 먼저 에러 메시지를 잘 읽어보아야 합니다. (코드를 다른 PR로 올려주셨다면 제가 볼 수 있었을 텐데요.)

- `Error creating bean with name 'portfolioRepository'`
  - `portfolioRepository` bean을 생성하는 과정에서 에러가 발생했습니다.
- `Cannot resolve reference to bean 'jpaMappingContext' while setting bean property 'mappingContext';`
  - `mappingContext` bean 속성을 셋팅할 때 `jpaMappingContext`bean의 레퍼런스를 resolve할 수 없음(찾지 못함).

`jpaMappingContext` bean을 못 찾고 있는 게 문제라고 볼 수 있습니다. 
DB는 띄워둔 상태에서 실행하셨는지요? postgresql 연동 전에 H2 에서는 잘 돌아갔나요?

tbnsok40 committed:
h2 DB 연결 상태

tbnsok40 commented:
> > postgresql 을 연동하는 과정에서 아래와 같은 에러가 발생하였습니다.
> > ` Error creating bean with name 'portfolioRepository' defined in project.spring.tbnsok40.domain.PortfolioRepository defined in @EnableJpaRepositories declared on JpaRepositoriesRegistrar.EnableJpaRepositoriesConfiguration: Cannot resolve reference to bean 'jpaMappingContext' while setting bean property 'mappingContext';`
> > 이해한 바로는, PortfolioRepository bean 을 생성하지 못해 발생한 에러입니다.
> > PortfolioRepository 를 사용하는 PortfolioService 에서 PortfolioRepository의 선언부에 @Autowired 어노테이션을 붙여봤지만 동일한 에러메시지가 발생하여, 에러의 다른 원인이 무엇인지 질문드리고 싶습니다..!
> 
> 이럴 땐 먼저 에러 메시지를 잘 읽어보아야 합니다. (코드를 다른 PR로 올려주셨다면 제가 볼 수 있었을 텐데요.)
> 
> * `Error creating bean with name 'portfolioRepository'`
>   
>   * `portfolioRepository` bean을 생성하는 과정에서 에러가 발생했습니다.
> * `Cannot resolve reference to bean 'jpaMappingContext' while setting bean property 'mappingContext';`
>   
>   * `mappingContext` bean 속성을 셋팅할 때 `jpaMappingContext`bean의 레퍼런스를 resolve할 수 없음(찾지 못함).
> 
> `jpaMappingContext` bean을 못 찾고 있는 게 문제라고 볼 수 있습니다.
> DB는 띄워둔 상태에서 실행하셨는지요? postgresql 연동 전에 H2 에서는 잘 돌아갔나요?

사실 postgresql 을 연동해야하기 때문에 h2에 대해서는 생각을 못했습니다.. 
피드백받고 우선 h2 DB 를 돌렸습니다, 제가 만든 entity가 스키마로 생성된것도 확인했됐습니다!

아직 postgresql 이 문제인데, 새로 풀리퀘를 생성하여 진행하겠습니다. 감사합니다 !
